server/api/users/users.controller.ts

```diff
@@ -29,9 +29,32 @@ export const getUserById: RequestHandler<{id: string}, UserProfile> = async (req
 };
 
 export const getUsers: RequestHandler<UserProfile[]> = async (req, res) => {
-  const users = await User.findAll({ attributes: User.allowedFields });
+  const page = req.query.page ? parseInt(req.query.page as string, 10) : 1;
+  const limit = 20;
+  const offset = (page - 1) * limit;
 
-  res.json(users);
+  if (!Number.isInteger(page) || page < 1) {
+    return res.status(400).json({ error: 'Invalid page number' });
+  }
+
+  // Fine all users, using the limit and offset to implement offset pagination
+  const users = await User.findAll({
+    attributes: User.allowedFields,
+    limit: limit,
+    offset: offset,
+  });
+
+  const count = await User.count();
+  const totalPages = Math.ceil(count / limit);
+
+  if (offset >= count && page !== 1) {
+    return res.status(400).json({ error: 'Page out of range' });
+  }
+
+  res.json({
+    page: page,
+    totalPages: totalPages,
+    users });
```

Comment:
nit:
```suggestion
  res.json({ page, totalPages, users });
```

Reply:
i was thinking we could add a linting rule for this since it happens a few times throughout this PR haha

Reply:
@timmyichen likewise, though I wasn't sure which settings to use. `always`? `avoidQuotes`? 
[ESLint Docs: object-shorthand](https://eslint.org/docs/latest/rules/object-shorthand)

I already have a branch with some other rules on it, I can add this one to it.

Reply:
Yup I think always + avoid Quotes is good!

Reply:
you nit and request to change linting has been DENIED